compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -1473,22 +1474,32 @@ impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
                         let ty_from = op.ty(self.body, tcx);
                         let cast_ty_from = CastTy::from_ty(ty_from);
                         let cast_ty_to = CastTy::from_ty(*ty);
+
                         match (cast_ty_from, cast_ty_to) {
                             (Some(CastTy::Ptr(src)), Some(CastTy::Ptr(dst))) => {
```

Comment:
while we're here, can you change this to `let Some(CastTy::Ptr(src)) = CastTy::from(ty_from) else { unreachable() };`

instead of matching on them? THis saves 2 levels of identation and it feels a bit hard to read rn